ingestion/functions/parsing/brazil_goias/goias.py

```diff
@@ -0,0 +1,189 @@
+    "diabetes": "diabetes mellitus"
+}
+
+
+def convert_date(raw_date):
+    """
+    Convert raw date field into a value interpretable by the dataserver.
+    """
+    date = datetime.strptime(raw_date, "%Y%m%d")
+    return date.strftime("%m/%d/%YZ")
+
+
+def convert_gender(raw_gender: str):
+    if raw_gender == "MASCULINO":
+        return "Male"
+    if raw_gender == "FEMININO":
+        return "Female"
+
+
+def convert_events(date_confirmed, date_symptoms, recovered):
+    events = [
+        {
+            "name": "confirmed",
+            "dateRange": {
+                "start": convert_date(date_confirmed),
+                "end": convert_date(date_confirmed),
+            },
+        }
+    ]
+    if date_symptoms:
+        events.append(
+            {
+                "name": "onsetSymptoms",
+                "dateRange": {
+                    "start": convert_date(date_symptoms),
+                    "end": convert_date(date_symptoms),
+                },
+            }
+        )
+    if recovered == "SIM":
+        events.append(
+            {
+                "name": "outcome",
+                "value": "Recovered",
+            }
+        )
+    return events
+
+
+def convert_preexisting_conditions(lung: str, cardiovascular: str, diabetes: str):
+    preexistingConditions = {}
+    items = (lung, cardiovascular, diabetes)
+    if all(item == "NAO" for item in items):
+        return None
+
+    preexistingConditions["hasPreexistingConditions"] = True
+    comorbidities = []
+
+    if lung == "SIM":
+        comorbidities.append(_COMORBIDITIES_MAP["doenca_respiratoria"])
+    if cardiovascular == "SIM":
+        comorbidities.append(_COMORBIDITIES_MAP["doenca_cardiovascular"])
+    if diabetes == "SIM":
+        comorbidities.append(_COMORBIDITIES_MAP["diabetes"])
+
+    if comorbidities:
+        preexistingConditions["values"] = comorbidities
+
+    return preexistingConditions
+
+
+def convert_ethnicity(ethnicity: str):
+    if ethnicity == "PRETA":
+        return "Black"
+    elif ethnicity == "PARDA":
+        return "Mixed"
+    elif ethnicity == "AMARELA":
+        return "Asian"
+    elif ethnicity == "BRANCA":
+        return "White"
+    elif ethnicity == "INDIGENA":
+        return "Indigenous"
+
+
+def convert_demographics(gender: str, age: str, ethnicity: str):
+    if not any((gender, age, ethnicity)):
+        return None
+    demo = {}
+    if gender:
+        demo["gender"] = convert_gender(gender)
+    if age:
+        if age == ">= 80 anos":
+            demo["ageRange"] = {"start": 80, "end": 120}
+        # No age resolution provided below 10 years
+        elif age == "< 10 anos":
+            demo["ageRange"] = {"start": 0, "end": 9}
+        else:
+            # Age in format '20 a 29 anos'
+            age_range = age.partition(" a ")
+            demo["ageRange"] = {"start": float(age_range[0]), "end": float("".join([i for i in age_range[2] if not i.isalpha()]))}
```

Comment:
For `age_range[2]`, is that ever not two characters? If it is always two characters, would `float(age_range[2][:2])` do the trick?

I love the use of partition, which I had not heard of before (had only used split).

Reply:
Nice one, so if I understand correctly age_range[2][:2] takes only the first two characters of e.g. '29 anos' ?

Reply:
Yes that's what it should do!